File: pset2 - crypto/caesar.c
// PBS @ CS50x
// 24.07.2017
// Implement a program that encrypts messages using Caesar’s cipher

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main( int argc, string argv[] )
{
    
    if( ! argv[1] ){
      
      printf("please insert your cypher key\n");
      return 1;
        
    }
    
    (void)argc;
    string input;
    int key = atoi( argv[1] );
    
    if( key < 0 )
    {
        
        printf("non-negative integers only\n");
        return 2;   
        
    }
    
    do{
        
        printf("plaintext: ");
        input = get_string();
        
    }while( input[0] == '\0' );
    
    int alphabetical_index;
    int remain;
    int ascii_start;
    
    printf("ciphertext: ");
    
    for( int i = 0, n = strlen( input ); i < n; i++ )
    {
        
        if( isalpha( input[i] ) )
        {
            
            // know where to start counting
            if( islower( input[i] ) )
                ascii_start = 97;
            else
                ascii_start = 65;
            
            //printf("we have a alpha with ASCCI value of %i ", input[i] );
            
            // calculates alpha index, example: "a" = 65 - 65 = 0
            alphabetical_index = input[i] - ascii_start;
            
            //printf("and alphabetical index of %i \n", alphabetical_index );
            
            // remainder of 26 ( chars ) that is the offset
            remain = ( alphabetical_index + key ) % 26;
            
            //printf("and the remain is %i \n", remain );
            
            // add the offset ( remain ) to ascii_start
            // to get the ascii number of the cyphered char
            printf("%c", ascii_start + remain );
            
        }
        else
        {
            // we have symbol or numeric char
            printf("%c", input[i]);
        }
        
    }
    
    printf("\n");
    
}

File: pset6 - sentimental/caesar.py
# PBS @CS50x
# 10.10.2017
# Implement a program that encrypts messages using Caesar’s cipher
# conversion from C

import cs50


def main():
    print("please insert your cypher key:")

    # get key
    while True:
        key = cs50.get_int()

        if(key < 0):
            print("non-negative integers only")
        else:
            break

    # get plain text
    while True:

        print("please insert plain text:")

        text = cs50.get_string()

        if(text == ""):
            print("please insert plain text:")
        else:
            break

    # ok, we got key and text
    # loop thru string chars..

    alphabetical_index = 0
    remain = 0
    ascii_start = 0

    for c in text:

        # print(c)

        # validate char as alphabetic
        if(c.isalpha()):

            # know where to start counting on ASCII table
            # if lower, starts at 97
            if(c.islower()):
                ascii_start = 97
            else:
                ascii_start = 65

            # print(ascii_start)

            # calculates alpha index, example: "a" = 65 - 65 = 0
            alphabetical_index = ord(c) - ascii_start

            # print(alphabetical_index)

            # remainder of 26 ( chars ) that is the offset
            remain = (alphabetical_index + key) % 26

            # print("the remain is %i \n", remain );

            # add the offset ( remain ) to ascii_start
            # to get the ascii number of the cyphered char
            print(chr(ascii_start + remain), end="")

        else:
            # we have symbol or numeric char
            print(c, end="")

    print("")


if __name__ == "__main__":
    main()
